Builds a shared liblua.so when LUA_SO is set, as nixpkgs does with lua-dso.make.

--- a/src/Makefile	2026-10-04 00:31:26.597555027 -0400
+++ b/src/Makefile	2026-10-04 00:31:26.602232767 -0400
@@ -44,7 +44,8 @@
 LUAC_O=	luac.o
 
 ALL_O= $(BASE_O) $(LUA_O) $(LUAC_O)
-ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T)
+LUA_SO=
+ALL_T= $(LUA_A) $(LUA_SO) $(LUA_T) $(LUAC_T)
 ALL_A= $(LUA_A)
 
 # Targets start here.
@@ -223,3 +224,9 @@
  lobject.h ltm.h lzio.h
 
 # (end of Makefile)
+
+ifneq ($(LUA_SO),)
+$(LUA_SO): $(CORE_O) $(LIB_O)
+	$(CC) -shared $(LIBS) -Wl,-soname,$(LUA_SO).5.4 -o $@.5.4 $? $(MYLDFLAGS)
+	ln -sf $(LUA_SO).5.4 $(LUA_SO)
+endif
